src/VisualStudio/Core/Def/Implementation/Snippets/AbstractSnippetInfoService.cs

```diff
@@ -111,52 +111,94 @@ public virtual bool ShouldFormatSnippet(SnippetInfo snippetInfo)
 
         private void PopulateSnippetCaches()
         {
-            var updatedSnippets = GetSnippetInfoList();
-            var updatedSnippetShortcuts = GetShortcutsHashFromSnippets(updatedSnippets);
+            Debug.Assert(_expansionManager != null);
 
-            lock (cacheGuard)
+            var token = _waiter.BeginAsyncOperation(GetType().Name + ".Start");
+            if (_expansionManager is IExpansionManager)
             {
-                snippets = updatedSnippets;
-                snippetShortcuts = updatedSnippetShortcuts;
+                // Call the asynchronous snippet population API on a background thread
+                Task.Factory.StartNew(async () => await PopulateSnippetCacheAsync().ConfigureAwait(false),
+                            CancellationToken.None,
+                            TaskCreationOptions.None,
+                            TaskScheduler.Default).CompletesAsyncOperation(token);
+            }
+            else
```

Comment:
IExpansionManager extends from IVsExpansionManager. Any future versions would also extend from IVsExpansionManager, so it will always be the valid fallback.